res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2149 @@
+
+PioneerDDJSX.loopMoveBackButton = function(channel, control, value, status, group) {
+    script.toggleControl(group, "loop_move_1_backward");
+};
+
+PioneerDDJSX.loopMoveForwardButton = function(channel, control, value, status, group) {
+    script.toggleControl(group, "loop_move_1_forward");
+};
+
+PioneerDDJSX.loadButton = function(channel, control, value, status, group) {
+    if (value) {
+        engine.setValue(group, "LoadSelectedTrack", true);
+        if (PioneerDDJSX.autoPFL) {
+            for (var index in PioneerDDJSX.channelGroups) {
+                if (PioneerDDJSX.channelGroups.hasOwnProperty(index)) {
+                    if (index === group) {
+                        engine.setValue(index, "pfl", true);
+                    } else {
+                        engine.setValue(index, "pfl", false);
+                    }
+                }
+            }
+        }
+    }
+};
+
+PioneerDDJSX.reverseRollButton = function(channel, control, value, status, group) {
+    script.toggleControl(group, "reverseroll");
+};
+
+PioneerDDJSX.reverseButton = function(channel, control, value, status, group) {
+    script.toggleControl(group, "reverse");
+};
+
+PioneerDDJSX.gridAdjustButton = function(channel, control, value, status, group) {
+    if (value) {
+        PioneerDDJSX.gridAdjustSelected = true;
+    } else {
+        PioneerDDJSX.gridAdjustSelected = false;
+    }
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.gridAdjust, value);
+};
+
+PioneerDDJSX.gridSetButton = function(channel, control, value, status, group) {
+    script.toggleControl(group, "beats_translate_curpos");
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.shiftGridAdjust, value);
+};
+
+PioneerDDJSX.gridSlideButton = function(channel, control, value, status, group) {
+    if (value) {
+        PioneerDDJSX.gridSlideSelected = true;
+    } else {
+        PioneerDDJSX.gridSlideSelected = false;
+    }
+    PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.gridSlide, value);
+};
+
+PioneerDDJSX.brakeButton = function(channel, control, value, status, group) {
+    script.brake(channel, control, value, status, group);
+};
+
+PioneerDDJSX.syncButton = function(channel, control, value, status, group) {
+    if (value) {
+        script.toggleControl(group, "sync_enabled");
+
+        PioneerDDJSX.syncRate = engine.getValue(group, "rate");
+
+        if (PioneerDDJSX.syncRate > 0) {
+            PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.takeoverMinus, 1);
+        } else if (PioneerDDJSX.syncRate < 0) {
+            PioneerDDJSX.nonPadLedControl(group, PioneerDDJSX.nonPadLeds.takeoverPlus, 1);
+        }
+    }
+};
+
+PioneerDDJSX.quantizeButton = function(channel, control, value, status, group) {
+    if (value) {
+        script.toggleControl(group, "quantize");
+    }
+};
+
+PioneerDDJSX.needleSearchTouched = false;
+
+PioneerDDJSX.needleSearchTouch = function(channel, control, value, status, group) {
```

Comment:
When does this get executed? When touching the needle search? If so, I'm a bit confused about the purpose of the signal. Or is there a button to hold down while using the touch strip to prevent accidents?